test/Streams/SeekableReader.test.cpp

```diff
@@ -0,0 +1,65 @@
+#include "Streams/MemoryReader.h"
+#include "Streams/FileReader.h"
+#include "Streams/FileSliceReader.h"
+#include <gtest/gtest.h>
+#include <array>
+
+template <class T>
+Stream::SeekableReader* CreateSeekableReader();
+
+template <>
+Stream::SeekableReader* CreateSeekableReader<Stream::MemoryReader>() {
+	std::array<char, 5> buffer{ 't', 'e', 's', 't', '!' };
+	return new Stream::MemoryReader(buffer.data(), buffer.size() * sizeof(char));
+}
```

Comment:
This should be added to MemoryReader.test.cpp.

Reply:
This takes some overhead by creating SeekableReader.test.h. I just tried and it seems to work.